Automattic/sensei
# Add course navigation extra icons

Fixes #4567

### Changes proposed in this Pull Request

* It adds icons/status for some extra cases in the Learning Mode sidebar.
  * It adds the preview and locked states.
  * For quiz failed and awaiting grade, we added the same icon that we use when the lesson is in progress (the half filled circle). cc @sruj09 if we need something different here.
* It refactors the current icons to use generic names, so it can be shared with other scopes (the case of the `lock` icon, for example, that we use in different parts of the code).
* It refactors some methods to be `private` since we weren't using in any other part.

### Testing instructions


* Enable the feature flag: `add_filter( 'sensei_feature_flag_course_theme', '__return_true' );`
* Create a course with lessons.
* Add a lesson prerequisite to some lessons, and enable preview for others (both set through lesson sidebar).
* Logged-out, check that the lessons contain the lock icon and the eye icon for the lessons with preview enabled.
* Logged-in, check that lessons with not completed prerequisite also have the lock icon.
* Add a quiz to a lesson, and make sure it shows the half filled circle when awaiting grande, and when failed).


### Screenshot / Video

<img width="1190" alt="Screen Shot 2022-01-03 at 19 06 51" src="https://user-images.githubusercontent.com/876340/147985664-1d97e9c0-bd70-4e23-9f62-2c64bacdfaf6.png">
